contracts/locked-astroport-vault/src/execute/basic_vault.rs

```diff
@@ -55,10 +59,13 @@ pub fn execute_deposit(
     }
     .into();
 
+    let state = STATE.load(deps.storage)?;
```

Comment:
why not load state at beginning of function instead of using mut deps?

Reply:
Because we want to load the state after the state has been updated, not before.